### PR TITLE
fixed float issue

### PR DESCRIPTION
**Pull request by joerivanarkel**

### CODE REVIEW

File: Interpreter/Values/Float.cs [hunk 1]
```diff
@@ -16,12 +16,12 @@ public Float(object value)
 
     public override BaseValue OperatedBy(Token _operator, BaseValue other)
     {
-        if (Value is not double && other.Value is not double)
+        if (Value is not float && other.Value is not float)
```
Comment (teusvanarkel) [inline Interpreter/Values/Float.cs:19]: override Value is maybe not necessary, try to use the base Value

File: Interpreter/Values/Variable.cs [hunk 2]
```diff
@@ -4,40 +4,48 @@
...
 
 public class Variable : BaseValue, IValue
 {
-    public IValueNode ValueNode { get; set; }
+    public IValueNode? ValueNode { get; set; }
     public override object Value { get ; set; }
     public ILogger Logger { get; set; }
 
-    public Variable(string value, ILogger logger)
+    public Variable(string value, ILogger logger, InterpreterFactory interpreterFactory)
```
Comment (teusvanarkel) [inline Interpreter/Values/Variable.cs:18]: Maybe move the logic to a method

File: Interpreter/Values/Variable.cs [hunk 3]
```diff
@@ -4,40 +4,48 @@
...
+        var ResultNode = SymbolTable.Instance(logger).Get(value);
+        if (ResultNode is not IValueNode)
         {
-            throw new NullReferenceException("Token provided has no value");
+           var interpreter = interpreterFactory.GetInterpreter(ResultNode, logger);
+           var result = interpreter.VisitNode();
+           Value = result.Value;
+        }
+        else
+        {
```
Comment (teusvanarkel) [inline Interpreter/Values/Variable.cs:28]: Remove duplicate call to the symbol table

File: Interpreter/Values/Variable.cs [hunk 4]
```diff
@@ -4,40 +4,48 @@
...
+    public IValueNode? ValueNode { get; set; }
     public override object Value { get ; set; }
     public ILogger Logger { get; set; }
 
-    public Variable(string value, ILogger logger)
+    public Variable(string value, ILogger logger, InterpreterFactory interpreterFactory)
     {
-        ValueNode = (IValueNode) SymbolTable.Instance(logger).Get(value);
-        if (ValueNode.Value.Value is null)
+        var ResultNode = SymbolTable.Instance(logger).Get(value);
```
Comment (teusvanarkel) [inline Interpreter/Values/Variable.cs:20]: Lower case

File: Interpreter/Values/Variable.cs [hunk 5]
```diff
@@ -4,40 +4,48 @@
...
 using Parser.Node.Interfaces;
+using Interpreter.Interpreters;
+using Parser.Node;
 
 namespace Interpreter.Values;
 
 public class Variable : BaseValue, IValue
 {
-    public IValueNode ValueNode { get; set; }
+    public IValueNode? ValueNode { get; set; }
```
Comment (teusvanarkel) [inline Interpreter/Values/Variable.cs:14]: Remove...not used

File: Interpreter/Values/Variable.cs [hunk 6]
```diff
@@ -4,40 +4,48 @@
...
+            case Type stringType when Value.GetType() == typeof(string):
                 return new String(Value, Logger).OperatedBy(_operator, other);
-            case TypeCode.Double:
+            case Type floatType when Value.GetType() == typeof(float):
                 return new Float(Value).OperatedBy(_operator, other);
-            case TypeCode.Char:
+            case Type charType when Value.GetType() == typeof(char):
                 return new Char(Value, Logger).OperatedBy(_operator, other);
         }
         Logger.Log("No TypeCode found", this.GetType().Name, Common.Enum.LogType.ERROR);
```
Comment (teusvanarkel) [inline Interpreter/Values/Variable.cs:48]: Removed logging just before error....log the error!

**Comment by joerivanarkel** [comment]
**Summary**:
- Moved Logger and Factory to BaseInterpeter
- Moved Logger and Value to BaseVaule
- Added more possible nodes to variables
- Started on Validation methods per entity. #60 
- Fixed Float and Character lexing errors

**Pull request merged**